exchange-application/frontend/v3/src/FetchUtils.js

```diff
@@ -1,7 +1,11 @@
 async function fetchWithRetry (url, options=null, retries=3, msTimeout=3000) {
     const timerPromise = createTimerPromise(msTimeout)
     try {
-        return await Promise.race([fetch(url, options), timerPromise])
+        const response = await Promise.race([fetch(url, options), timerPromise]);
+        if (!isValid(response)) {
```

Comment:
Why not use the `response.ok`? Did it not work? (would be easier to understand imho)

Reply:
Yeah I didn't know about this one. Much cleaner. Fixed